### PR TITLE
Disable tls/617

### PR DESCRIPTION
**Pull request by rigazilla**
Fix for #617

### CODE REVIEW

File: pkg/apis/infinispan/v1/infinispan_types.go [hunk 1]
```diff
@@ -33,6 +33,11 @@ const (
 	CertificateSourceTypeSecret CertificateSourceType = "Secret"
 	// CertificateSourceTypeSecretLowCase certificate coming from a user provided secret
 	CertificateSourceTypeSecretLowCase CertificateSourceType = "secret"
+
+	// CertificateSourceTypeNoneNoEncryption no certificate encryption disabled
+	CertificateSourceTypeNoneNoEncryption CertificateSourceType = "None"
```
Comment (dmvolod) [inline pkg/apis/infinispan/v1/infinispan_types.go:38]: @rigazilla I think we don't need lowercase for the new parameters. Both stiles was for migration case only

File: pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go [hunk 2]
```diff
@@ -5,9 +5,56 @@
...
 )
 
 func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
-	return map[string]common.OpenAPIDefinition{}
+	return map[string]common.OpenAPIDefinition{
+		"github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v2alpha1.Backup": schema_pkg_apis_infinispan_v2alpha1_Backup(ref),
+	}
+}
+
+func schema_pkg_apis_infinispan_v2alpha1_Backup(ref common.ReferenceCallback) common.OpenAPIDefinition {
```
Comment (dmvolod) [inline pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go:18]: @rigazilla it seems to be you are generated code with different SDK version

File: pkg/apis/infinispan/v1/infinispan_types.go (hunk 1 above)
Comment (rigazilla) [reply pkg/apis/infinispan/v1/infinispan_types.go:38]: @dmvolod great! I've forgot about it

File: pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go (hunk 2 above)
Comment (rigazilla) [reply pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go]: no SDK is ok.  I've run an unneeded `generate openapi`. thanks

Comment (dmvolod) [reply pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go]: Oh, yes. I'm prefer to type just 'make codegen` operator-sdk ... to long for me :smile:

**Comment by dmvolod** [comment]
@rigazilla I'm not sure, but we should change something on the Router side
Have you check this case?

File: pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go (hunk 2 above)
Comment (rigazilla) [reply pkg/apis/infinispan/v2alpha1/zz_generated.openapi.go]: ouch! you're right!

**Comment by rigazilla** [comment]
> @rigazilla I'm not sure, but we should change something on the Router side
> Have you check this case?

yeah, @dmvolod always precious reviews :+1: 
tls needs to be disabled to do proper routing

**Review by dmvolod — APPROVED** [review]

File: pkg/apis/infinispan/v1/types_util.go [hunk 3]
```diff
@@ -316,6 +316,12 @@ func (ispn *Infinispan) IsUpgradeNeeded(logger logr.Logger) bool {
 	return false
 }
 
+// IsEncryptionDisabledByConfig returns true if encryption is disable by configuration
+func (ispn *Infinispan) IsEncryptionDisabledByConfig() bool {
```
Comment (ryanemerson) [inline pkg/apis/infinispan/v1/types_util.go:320]: We can simplify this to just `IsEncryptionDisabled`.

**Review by ryanemerson — COMMENTED** [review]
@rigazilla Needs a rebase on master and squashing.

**Comment by ryanemerson** [comment]
Thanks  @rigazilla

**Pull request merged**